src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -1234,19 +1238,30 @@ inline void rpp_load24_f32pln3_to_f32pln3_mirror_avx(Rpp32f *srcPtrR, Rpp32f *sr
 
 inline void rpp_store24_f32pln3_to_f32pkd3_avx(Rpp32f *dstPtr, __m256 *p)
 {
+    __m256 pTemp[4], pRow[4];
+    pTemp[0] = _mm256_shuffle_ps(p[0], p[1], 0x44);
+    pTemp[2] = _mm256_shuffle_ps(p[0], p[1], 0xEE);
+    pTemp[1] = _mm256_shuffle_ps(p[2], avx_p0, 0x44);
+    pTemp[3] = _mm256_shuffle_ps(p[2], avx_p0, 0xEE);
+    pRow[0] = _mm256_shuffle_ps(pTemp[0], pTemp[1], 0x88);
+    pRow[1] = _mm256_shuffle_ps(pTemp[0], pTemp[1], 0xDD);
+    pRow[2] = _mm256_shuffle_ps(pTemp[2], pTemp[3], 0x88);
+    pRow[3] = _mm256_shuffle_ps(pTemp[2], pTemp[3], 0xDD);
+
     __m128 p128[4];
-    p128[0] = _mm256_extractf128_ps(p[0], 0);
-    p128[1] = _mm256_extractf128_ps(p[1], 0);
-    p128[2] = _mm256_extractf128_ps(p[2], 0);
-    _MM_TRANSPOSE4_PS(p128[0], p128[1], p128[2], p128[3]);
+    p128[0] = _mm256_castps256_ps128(pRow[0]);
```

Comment:
Please add some inline comments for better readability on vectorized code

Reply:
Done